File: Prog2-9302-AY225-Salvador/MIDTERM-LAB-3/MP13/MP13.java
import java.io.*;
import java.util.*;

public class MP13 {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter CSV file path: ");
        String filePath = input.nextLine();

        try (BufferedReader br = new BufferedReader(new FileReader(filePath))) {
            String line;
            int rowNum = 0;

            while ((line = br.readLine()) != null) {
                rowNum++;
                String[] values = line.split(",", -1); // keep empty cells
                boolean hasMissing = false;
                for (String val : values) {
                    if (val.trim().isEmpty()) {
                        hasMissing = true;
                        break;
                    }
                }

                if (hasMissing) {
                    System.out.print("Row " + rowNum + ": ");
                    System.out.println(String.join(" | ", values));
                }
            }

        } catch (FileNotFoundException e) {
            System.out.println("File not found: " + filePath);
        } catch (IOException e) {
            System.out.println("Error reading file: " + e.getMessage());
        }
    }
}

File: Prog2-9302-AY225-Salvador/MIDTERM-LAB-3/MP13/MP13.js
const fs = require('fs');
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

rl.question("Enter CSV file path: ", function(filePath) {
    try {
        const data = fs.readFileSync(filePath, "utf8");
        const lines = data.trim().split("\n");

        lines.forEach((line, index) => {
            const values = line.split(",");
            const hasMissing = values.some(v => v.trim() === "");
            if (hasMissing) {
                console.log(`Row ${index + 1}: ${values.join(" | ")}`);
            }
        });

    } catch (err) {
        console.log("Error reading file: " + err.message);
    }
    rl.close();
});